lib/api/include/irods/ticketAPI.hpp

```diff
@@ -0,0 +1,414 @@
+#ifndef IRODS_TICKET_API_HPP
+#define IRODS_TICKET_API_HPP
+
+#include <string_view>
+#include <type_traits>
+#include <iostream>
+#include <string>
+
+#undef NAMESPACE_IMPL
+#undef RxComm
+#undef rxTicketAdmin
+
+#ifdef IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+#    define NAMESPACE_IMPL server
+#    define RxComm         RsComm
+#    define rxTicketAdmin  rsTicketAdmin
+
+struct RsComm;
+#else
+#    define NAMESPACE_IMPL client
+#    define RxComm         RcComm
+#    define rxTicketAdmin  rcTicketAdmin
+
+struct RcComm;
+#endif // IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+/**
+ * @brief Namespace to contain the API for tickets
+ *
+ */
+namespace irods::administration::ticket
+{
+    inline struct admin_tag
+    {
+    } admin;
+
+    enum class ticket_type
+    {
+        read,
+        write
+    };
+
+    enum class ticket_property
+    {
+        uses_count,
+        write_file,
+        write_byte
+    };
+
+    void execute_ticket_operation(RxComm& conn,
+                                  std::string_view command,
+                                  std::string_view ticket_identifier,
+                                  std::string_view command_modifier1,
+                                  std::string_view command_modifier2,
+                                  std::string_view command_modifier3,
+                                  std::string_view command_modifier4,
+                                  bool run_as_admin);
```

Comment:
This needs to be moved to the cpp file and put inside of an anonymous namespace.

Developers should not have access to this. If we can hide things like this, they won't attempt to use it.

Reply:
Discussed that putting this in its own namespace, detail, is solution. Will change it soon

Reply:
@alanking Please mark this as resolved.